misk/src/test/kotlin/misk/backoff/RetryTest.kt

```diff
@@ -0,0 +1,67 @@
+package misk.backoff
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Assertions.assertThrows
+import org.junit.jupiter.api.Test
+import java.time.Duration
+
+internal class RetryTest {
+  @Test fun retries() {
+    val backoff = ExponentialBackoff(Duration.ofMillis(10), Duration.ofMillis(100))
+
+    val result = retry(3, backoff) { retry ->
```

Comment:
i'm surprised the function is not passed as a parameter. definitely some KT thing I don't understand.

Reply:
If the final parameter to a function is itself a function, you can use this alternate syntax.

Reply:
ok. that makes sense to exit